### PR TITLE
[Refactor] Refactor the images command flagging process

### PR DESCRIPTION
**Pull request by Zheaoli**
Checklist:

- [x] Create a file in `pkg/api/types/${cmd}_types.go`, and define the CommandOption for this command
- [x] Create some file in `pkg/cmd/${cmd}`, and move the command entry point in real into this package

Signed-off-by: Zheao.Li <me@manjusaka.me>

### CODE REVIEW

**Review by AkihiroSuda — APPROVED** [review]
Thanks

File: cmd/nerdctl/images.go [hunk 1]
```diff
@@ -89,348 +66,72 @@ Properties:
...
+	return types.ImagesCommandOptions{
+		GOptions:         *globalOptions,
+		Quiet:            quiet,
+		NoTrunc:          noTrunc,
+		Format:           format,
+		Filters:          inputFilters,
+		NameAndRefFilter: filters,
+		Digests:          digests,
+		Names:            names,
+		All:              all,
```
Comment (AkihiroSuda) [inline cmd/nerdctl/images.go:124]: `All` should be always set to `true`?

File: pkg/cmd/images/images.go [hunk 2]
```diff
@@ -0,0 +1,407 @@
...
+	"github.com/containerd/nerdctl/pkg/api/types"
+	"github.com/containerd/nerdctl/pkg/clientutil"
+	"github.com/containerd/nerdctl/pkg/formatter"
+	"github.com/containerd/nerdctl/pkg/imgutil"
+	"github.com/opencontainers/image-spec/identity"
+	v1 "github.com/opencontainers/image-spec/specs-go/v1"
+	"github.com/sirupsen/logrus"
+)
+
+func Images(ctx context.Context, options types.ImagesCommandOptions) error {
```
[the remark this answers is not in the document]
Comment (Zheaoli) [reply pkg/cmd/images/images.go:49]: updated

**Comment by AkihiroSuda** [comment]
lint is failing

File: pkg/cmd/images/images.go [hunk 3]
```diff
@@ -0,0 +1,407 @@
+/*
```
Comment (AkihiroSuda) [inline pkg/cmd/images/images.go:1]: Probably this should be moved to `pkg/cmd/image/list.go`.
`pkg/cmd/load` should be merged into `pkg/cmd/image` too. (Can be another PR)

File: cmd/nerdctl/ps.go [hunk 4]
```diff
@@ -385,3 +387,57 @@ func getContainerSize(ctx context.Context, client *containerd.Client, c containe
 
 	return fmt.Sprintf("%s (virtual %s)", progress.Bytes(containerSize).String(), progress.Bytes(imageSize).String()), nil
 }
+
+// TODO: this code should be remove to a common package after the refactor completed
+type snapshotKey string
```
Comment (djdongjin) [inline cmd/nerdctl/ps.go:392]: Is this `snapshotKey` code the same as the code in `pkg/cmd/image/list.go`

File: pkg/cmd/image/list.go [hunk 5]
```diff
@@ -0,0 +1,407 @@
...
+	Size         string // the size of the unpacked snapshots.
+	BlobSize     string // the size of the blobs in the content store (nerdctl extension)
+	// TODO: "SharedSize", "UniqueSize", "VirtualSize"
+	Platform string // nerdctl extension
+}
+
+func printImages(ctx context.Context, options types.ImagesCommandOptions, client *containerd.Client, imageList []images.Image) error {
+	digestsFlag := options.Digests
+	var w io.Writer
+	w = os.Stdout
```
Comment (djdongjin) [inline pkg/cmd/image/list.go:187]: I know this is original code, but later we might pass a writer to this func instead of binding to os.Stdout

File: cmd/nerdctl/ps.go (hunk 4 above)
Comment (Zheaoli) [reply cmd/nerdctl/ps.go:392]: Yes, the duplicate code will be remove when I refactor the ps command

File: pkg/cmd/image/list.go (hunk 5 above)
Comment (Zheaoli) [reply pkg/cmd/image/list.go:187]: SGTM, maybe another PR could be better

**Review by djdongjin — APPROVED** [review]

File: pkg/cmd/image/list.go [hunk 6]
```diff
@@ -0,0 +1,407 @@
...
+	"github.com/containerd/nerdctl/pkg/api/types"
+	"github.com/containerd/nerdctl/pkg/clientutil"
+	"github.com/containerd/nerdctl/pkg/formatter"
+	"github.com/containerd/nerdctl/pkg/imgutil"
+	"github.com/opencontainers/image-spec/identity"
+	v1 "github.com/opencontainers/image-spec/specs-go/v1"
+	"github.com/sirupsen/logrus"
+)
+
+func Images(ctx context.Context, options types.ImagesCommandOptions) error {
```
Comment (AkihiroSuda) [inline pkg/cmd/image/list.go:49]:
```suggestion
func ImageList(ctx context.Context, options types.ImagesCommandOptions) error {
```

File: pkg/api/types/images_type.go [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package types
+
+type ImagesCommandOptions struct {
```
Comment (AkihiroSuda) [inline pkg/api/types/images_type.go:19]:
```suggestion
type ImageListCommandOptions struct {
```
